src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -188,8 +189,9 @@ public bool TryGetTelemetryId(out Guid telemetryId)
                 {
                     using (operationContext?.AddScope(allowCancellation: true, description: EditorFeaturesWpfResources.Gathering_Suggestions_Waiting_for_the_solution_to_fully_load))
                     {
+                        using var combinedCancellation = (operationContext?.UserCancellationToken ?? CancellationToken.None).CombineWith(cancellationToken);
```

Comment:
Took a look at this - it does not fix the issue.  Note that the editor is calling into us from https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs#L162 (with the IUIThreadOperationContext only).  So the cancellationToken var here is actually the same as the operationContext?.UserCancellationToken.

Also note that the string shown in the wait dialog is "please wait for an editor command to finish" which is not the context description / title we provide.  I'm wondering if someone higher up the stack is not respecting cancellation.

Reply:
posted my theory here - https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1096969

Reply:
> which passes the IUIThreadOperationContext.UserCancellationToken as the cancellationToken

This token changes from time to time.